istioctl/cmd/add-to-mesh.go

```diff
@@ -492,7 +493,7 @@ func generateServiceEntry(u *unstructured.Unstructured, o *vmServiceOpts) error
 			Labels:  o.Labels,
 		})
 	}
-	host := fmt.Sprintf("%v.%v.svc.cluster.local", o.Name, o.Namespace)
+	host := fmt.Sprintf("%v.%v.svc.%s", o.Name, o.Namespace, constants.DefaultKubernetesDomain)
```

Comment:
Is it really a constant ? I think it is customizable - something in mesh config ?

Reply:
its not, but this PR doesn't change it from variable -> constant at least

Reply:
istio-agent seems to have this - I think we have portions of the code that just use cluster.local ignoring the setting, we should grep for cluster.local and maybe add an integration test where we create a cluster with a different domain...

Reply:
Agreed ... it's pretty inconsistent now. That was actually the main motivation with making it a constant here - at least we can identify all the places using it as a constant now. I've raised https://github.com/istio/istio/issues/23652 to address the broader issue.